reconstruction.py

```diff
@@ -493,27 +521,31 @@ def reconstruct(self,evrange=(-1,-1,-1)):
                     elif name.startswith('DGH0'):
                         pmt_read = True
                         exist_pmt = True
+                        fast_digitizer = False
+                        slow_digitizer = False
                         if self.options.pmt_mode:
                             header=cy.daq_dgz_full2header(mevent.banks[key], verbose=False)
-                            sample_rate = header.sampling_rate
-
-                            nChannels_f  = header.nchannels[0]
-                            nTriggers_f = len(header.TTT[0])
-                            TTTs_f = header.TTT[0]
-
-                            nChannels_s  = header.nchannels[1]
-                            nTriggers_s = len(header.TTT[1])
-                            TTTs_s = header.TTT[1]
+                            # sample_rate = header.sampling_rate
 
                             ## Care: if tag is MC$blabla, the tag for the digitizer will have to be changed to LNGS or something
                             waveform_f, waveform_s = cy.daq_dgz_full2array(mevent.banks['DIG0'], header, verbose=False, corrected=corrected, ch_offset=channels_offsets,tag=self.options.tag)
 
+                            if len(waveform_f):
```

Comment:
This could be generalised more, no? You have the header, so like in the cygno libs you could run over the boards with enumerate and read from header.nchannels[idigi] instead of header.nchannels[0 or 1].
Even more, you could test with the PMT board name and only then decide which digitizer flag needs to be true (slow or fast). 
in the future maybe this should become a list of waveforms and be further generalised, but for now ok.. do you agree?

Reply:
Ok, I generalized the script similarly to the cygno libs. 
I also performed tests with all detectors and readout modes and compared the trees.
Ok, I incorporated the board name check although it is a redundancy since `waveform_f` and `waveform_s` come associated with a given digitizer from the cygno libs by construction.
Yes, this part of the PMT/Digitizer readout is a bit messy and should be simplified and generalized to eventually receive the CYGNO_04 data. Probably we can add the header to the waveforms, or a different object that contains all the info needed.